### PR TITLE
Improve contacts and re-apply ordering

### PR DESCRIPTION
**Pull request by Shramp**
- Improve contacts handling logic and fix contact-related bugs that were introduced with the migration to FS 2 API
  - tx logs in DW used to have either and `assignedAddressId` or a `recipientAddressId`. I simplified this so that tx logs now just have an `address` field. We can match the logs to contacts because we know that received tx logs could match the `assignedAddress` field on a contact, and sent tx logs could match the `recipientAddressId` on a contact.
  - DW wallet used to use both `assignedAddressId` and `recipientAddressId` as identifiers for selecting and handling contacts. This was confusing, so I added an `id` field to the contact type (using uuid to populate it), and use that field as an identifier instead.

- Add back the alphabetizing eslint rules
  - this should be in a separate PR, but I started doing it in this one before committing the contacts code and then just continued. Makes this PR a little harder to look at, so I went through and commented on each file that contains only alphabetizing changes, so you don't have to look hard at em.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -38,6 +38,10 @@ module.exports = {
     'import/order': [
       'error',
       {
+        alphabetize: {
```
Comment (Shramp) [inline .eslintrc.js:41]: Decided that it was really nice having everything alphabetized after all

File: app/components/MOBNumberFormat/MOBNumberFormat.test.tsx [hunk 2]
```diff
@@ -3,8 +3,8 @@ import React from 'react';
 import { screen, render } from '@testing-library/react';
```
Comment (Shramp) [inline app/components/MOBNumberFormat/MOBNumberFormat.test.tsx:3]: just ordering changes in this file

File: app/components/MOBNumberFormat/MOBNumberFormat.view.tsx [hunk 3]
```diff
@@ -46,12 +46,12 @@ const MOBNumberFormat = (props: MOBNumberFormatProps): JSX.Element => {
 };
```
Comment (Shramp) [inline app/components/MOBNumberFormat/MOBNumberFormat.view.tsx:46]: just ordering changes in this file

File: app/constants/tokens.tsx [hunk 4]
```diff
@@ -22,18 +22,18 @@ function renderEUSDIcon({ color: _, ...rest }: IconProps): JSX.Element {
...
-    icon: renderMobIcon,
-    name: 'MOB',
-  },
   EUSD: {
     // TODO for testing we are using 8192 instead of 1
     // FIX-ME BEFORE MERGING INTO DEV, CHANGE THIS BACK TO 1
-    id: 8192,
-    precision: 1_000_000,
     icon: renderEUSDIcon,
+    id: 8192,
```
Comment (Shramp) [inline app/constants/tokens.tsx:29]: just ordering changes in this file

File: app/fullService/api/getBalanceForAccount.ts [hunk 5]
```diff
@@ -35,6 +31,10 @@ export function convertBalanceFromV2(accountStatus: AccountStatus): BalanceStatu
         unspentPmob: accountStatus.balancePerToken[TOKENS.EUSD.id]?.unspent || '0',
       },
     },
+    isSynced:
```
Comment (Shramp) [inline app/fullService/api/getBalanceForAccount.ts:34]: just ordering changes in this file

File: app/fullService/api/getWalletStatus.ts [hunk 6]
```diff
@@ -39,6 +34,11 @@ function convertWalletStatusFromV2(
           unspentPmob: status.balancePerToken[TOKENS.EUSD.id]?.unspent || '0',
         },
       },
+      isSyncedAll: status.isSyncedAll,
```
Comment (Shramp) [inline app/fullService/api/getWalletStatus.ts:37]: just ordering changes in this file

File: app/layouts/DashboardLayout/BalanceIndicator.view/BalanceIndicator.view.tsx [hunk 7]
```diff
@@ -14,12 +14,12 @@ import {
 import { useTranslation } from 'react-i18next';
 
 import { MOBNumberFormat } from '../../../components';
-import { TOKENS } from '../../../constants/tokens';
 import { GOLD_LIGHT } from '../../../constants/colors';
+import { TOKENS } from '../../../constants/tokens';
```
Comment (Shramp) [inline app/layouts/DashboardLayout/BalanceIndicator.view/BalanceIndicator.view.tsx:18]: just ordering changes in this file

File: app/layouts/DashboardLayout/SetPinModal.view/SetPinModal.view.tsx [hunk 8]
```diff
@@ -20,13 +20,13 @@ import * as Yup from 'yup';
 import { MOBNumberFormat, SubmitButton } from '../../../components';
 import { MOBIcon } from '../../../components/icons';
 import { PIN_MIN_SIZE } from '../../../constants/codes';
+import { useCurrentToken } from '../../../hooks/useCurrentToken';
```
Comment (Shramp) [inline app/layouts/DashboardLayout/SetPinModal.view/SetPinModal.view.tsx:23]: just ordering changes in this file

File: app/pages/Dashboard/DashboardPage.view/DashboardPage.test.tsx [hunk 9]
```diff
@@ -35,6 +31,10 @@ const selectedAccount = {
         unspentPmob: '908298888888888',
       },
     },
+    isSynced: true,
```
Comment (Shramp) [inline app/pages/Dashboard/DashboardPage.view/DashboardPage.test.tsx:34]: just ordering changes in this file

File: app/pages/Gifts/BuildGiftForm.view/BuildGiftForm.view.tsx [hunk 10]
```diff
@@ -22,10 +22,10 @@ import * as Yup from 'yup';
 import { MOBNumberFormat } from '../../../components/MOBNumberFormat';
 import { SubmitButton } from '../../../components/SubmitButton';
 import { MOBIcon } from '../../../components/icons';
+import { useCurrentToken } from '../../../hooks/useCurrentToken';
```
Comment (Shramp) [inline app/pages/Gifts/BuildGiftForm.view/BuildGiftForm.view.tsx:25]: just ordering changes in this file

File: app/pages/Gifts/BuildGiftPanel.view/BuildGiftPanel.test.tsx [hunk 11]
```diff
@@ -62,6 +58,10 @@ const SELECTED_ACCOUNT = {
         unspentPmob: '908298888888888',
       },
     },
+    isSynced: true,
```
Comment (Shramp) [inline app/pages/Gifts/BuildGiftPanel.view/BuildGiftPanel.test.tsx:61]: just ordering changes in this file

File: app/pages/Gifts/BuildGiftPanel.view/BuildGiftPanel.view.tsx [hunk 12]
```diff
@@ -28,9 +28,9 @@ import { useTranslation } from 'react-i18next';
 import { AccountCard, MOBNumberFormat } from '../../../components';
 import { ShortCode } from '../../../components/ShortCode';
 import { CopyIcon, TrashcanIcon } from '../../../components/icons';
+import { useCurrentToken } from '../../../hooks/useCurrentToken';
```
Comment (Shramp) [inline app/pages/Gifts/BuildGiftPanel.view/BuildGiftPanel.view.tsx:31]: just ordering changes in this file

File: app/pages/Gifts/ConsumeGiftForm.view/ConsumeGiftForm.test.tsx [hunk 13]
```diff
@@ -49,6 +45,10 @@ const selectedAccount = {
         unspentPmob: '908298888888888',
       },
     },
+    isSynced: true,
```
Comment (Shramp) [inline app/pages/Gifts/ConsumeGiftForm.view/ConsumeGiftForm.test.tsx:48]: just ordering changes in this file

File: app/pages/Gifts/ConsumeGiftForm.view/ConsumeGiftForm.view.tsx [hunk 14]
```diff
@@ -21,9 +21,9 @@ import { useTranslation } from 'react-i18next';
 import * as Yup from 'yup';
 
 import { SubmitButton, MOBNumberFormat } from '../../../components';
+import { useCurrentToken } from '../../../hooks/useCurrentToken';
```
Comment (Shramp) [inline app/pages/Gifts/ConsumeGiftForm.view/ConsumeGiftForm.view.tsx:24]: just ordering changes in this file

File: app/pages/Gifts/GiftsPage.presenter/GiftsPage.tsx [hunk 15]
```diff
@@ -19,6 +19,8 @@ import { useSelector } from 'react-redux';
 
 import { AccountCard } from '../../../components/AccountCard';
 import { SubmitButton } from '../../../components/SubmitButton';
+import { TOKENS } from '../../../constants/tokens';
+import { useCurrentToken } from '../../../hooks/useCurrentToken';
```
Comment (Shramp) [inline app/pages/Gifts/GiftsPage.presenter/GiftsPage.tsx:23]: just ordering changes in this file

File: app/pages/History/HistoryItem.view/HistoryItem.test.tsx [hunk 16]
```diff
@@ -32,8 +32,8 @@ describe('History Item', () => {
               direction: 'tx_direction_sent',
               finalizedBlockIndex: '123456',
               recipientAddressId: null,
-              value: '220960000000',
               tokenId: 0,
+              value: '220960000000',
```
Comment (Shramp) [inline app/pages/History/HistoryItem.view/HistoryItem.test.tsx:36]: just ordering changes in this file

File: app/pages/History/HistoryList.view/HistoryList.test.tsx [hunk 17]
```diff
@@ -24,19 +24,19 @@ describe('History list', () => {
               direction: 'tx_direction_sent',
               finalizedBlockIndex: '123456',
               recipientAddressId: null,
+              tokenId: 0,
```
Comment (Shramp) [inline app/pages/History/HistoryList.view/HistoryList.test.tsx:27]: just ordering changes in this file

File: app/pages/History/HistoryPage.presenter/HistoryPage.presenter.tsx [hunk 18]
```diff
@@ -10,13 +10,13 @@ import { Redirect } from 'react-router-dom';
 
 import { LoadingScreen } from '../../../components';
 import { getConfirmations, validateConfirmation } from '../../../fullService/api';
+import { useTransactionLogs } from '../../../hooks/useTransactionLogs';
```
Comment (Shramp) [inline app/pages/History/HistoryPage.presenter/HistoryPage.presenter.tsx:13]: just ordering changes in this file

File: app/pages/SendReceive/PaymentRequests.view/PaymentRequest.view.tsx [hunk 19]
```diff
@@ -21,12 +21,12 @@ import * as Yup from 'yup';
 
 import { SubmitButton, MOBNumberFormat } from '../../../components';
 import { LongCode } from '../../../components/LongCode';
+import { useCurrentToken } from '../../../hooks/useCurrentToken';
```
Comment (Shramp) [inline app/pages/SendReceive/PaymentRequests.view/PaymentRequest.view.tsx:24]: just ordering changes in this file

File: app/pages/SendReceive/SendMob.view/SendMob.test.tsx [hunk 20]
```diff
@@ -83,6 +79,10 @@ const SELECTED_ACCOUNT = {
         unspentPmob: String(INITIAL_BALANCE * 1000000000000),
       },
     },
+    isSynced: true,
```
Comment (Shramp) [inline app/pages/SendReceive/SendMob.view/SendMob.test.tsx:82]: just ordering changes in this file

File: app/pages/Settings/ChangePin.view/ChangePin.view.tsx [hunk 21]
```diff
@@ -19,11 +19,11 @@ import * as Yup from 'yup';
 import { SubmitButton, MOBNumberFormat, SavedPasswordsModal } from '../../../components';
 import { MOBIcon } from '../../../components/icons';
 import { PIN_MIN_SIZE } from '../../../constants/codes';
+import { useCurrentToken } from '../../../hooks/useCurrentToken';
```
Comment (Shramp) [inline app/pages/Settings/ChangePin.view/ChangePin.view.tsx:22]: just ordering changes in this file

File: app/redux/reducers/reducers.ts [hunk 22]
```diff
@@ -153,6 +149,10 @@ export const initialReduxStoreState: ReduxStoreState = {
         unspentPmob: '',
       },
     },
+    isSyncedAll: false,
```
Comment (Shramp) [inline app/redux/reducers/reducers.ts:152]: just ordering changes in this file

File: locales/enUS/translation.json [hunk 23]
```diff
@@ -612,4 +612,4 @@
     "title": "Unlock Wallet",
     "unlockWalletButton": "Unlock Wallet"
   }
-}
\ No newline at end of file
+}
```
Comment (Shramp) [inline locales/enUS/translation.json:615]: just ordering changes in this file

**Review by david1alvarez — APPROVED** [review]
LGTM, love the code cleanups!

**Pull request merged**